### ISSUE TITLE
IPA server installation fails with  @389ds/389-ds-base-nightly 

### ISSUE THREAD

**Issue by flo-renaud** [389ds/389-ds-base#5821]
**Issue Description**
IPA server iinstallation fails when using 389-ds packages from  @389ds/389-ds-base-nightly.

**Package Version and Platform:**
 - Platform: Fedora 38
 - Package and version: 389-ds-base-2.4.1-202307020400gitf2c1e44b5.fc38.x86_64

**Steps to Reproduce**
1. dnf copr enable -y @389ds/389-ds-base-nightly 
2. dnf install -y freeipa-server
3. ipa-server-install --domain ipa.test --realm IPA.TEST -a Secret123 -p Secret123 -U

**Expected results**
Server installation should complete successfully

**Additional context**
ipa-server-install fails in the final steps with a crash a ns-slapd after the SIDgen task has been started:
```
  [8/8]: adding SIDs to existing users and groups
This step may take considerable amount of time, please wait..
Exception occured during SID generation: cannot connect to 'ldapi://%2Frun%2Fslapd-IPA-TEST.socket': Connection reset by peer
Done.
cannot connect to 'ldapi://%2Frun%2Fslapd-IPA-TEST.socket':
```

The [journal](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f9fd7672-184e-11ee-933e-fa163ec53c61/test_integration-test_upgrade.py-TestUpgrade-install/master.ipa.test/journal.gz) shows a crash of ns-slapd:
```
                                                         #0  0x00007fdd603514b3 op_stat_destructor (libslapd.so.0 + 0x1514b3)
                                                         #1  0x00007fdd60320643 factory_destroy_extension (libslapd.so.0 + 0x120643)
                                                         #2  0x00007fdd6034edf4 operation_done (libslapd.so.0 + 0x14edf4)
                                                         #3  0x00007fdd6034ef16 operation_free (libslapd.so.0 + 0x14ef16)
                                                         #4  0x00007fdd60351974 pblock_done (libslapd.so.0 + 0x151974)
                                                         #5  0x00007fdd60351a5a slapi_pblock_destroy (libslapd.so.0 + 0x151a5a)
                                                         #6  0x00007fdd603649a1 slapi_search_internal_get_entry (libslapd.so.0 + 0x1649a1)
                                                         #7  0x00007fdd5e08ca4b ipauuid_pre_op (libipa_uuid.so + 0x3a4b)
                                                         #8  0x00007fdd6035f3fc plugin_call_func.lto_priv.0 (libslapd.so.0 + 0x15f3fc)
                                                         #9  0x00007fdd6035f759 plugin_call_plugins (libslapd.so.0 + 0x15f759)
                                                         #10 0x00007fdd6034cb1d op_shared_modify (libslapd.so.0 + 0x14cb1d)
                                                         #11 0x00007fdd6034e70e modify_internal_pb (libslapd.so.0 + 0x14e70e)
                                                         #12 0x00007fdd60394c7c slapi_task_status_changed.part.0 (libslapd.so.0 + 0x194c7c)
                                                         #13 0x00007fdd5e0730e1 sidgen_task_thread (libipa_sidgen_task.so + 0x40e1)
                                                         #14 0x00007fdd600ae907 start_thread (libc.so.6 + 0x8c907)
                                                         #15 0x00007fdd60134870 __clone3 (libc.so.6 + 0x112870)
```

The coredump contains the following info:
```
(gdb) where
#0  0x00007f319e5514b3 in op_stat_destructor (extension=<optimized out>, object=<optimized out>, parent=<optimized out>) at ldap/servers/slapd/operation.c:712
#1  0x00007f319e520643 in factory_destroy_extension (extension=0x7f3198fface0, parent=0x0, object=0x7f3198ffac00, type=<optimized out>) at ldap/servers/slapd/factory.c:366
#2  factory_destroy_extension (type=<optimized out>, object=0x7f3198ffac00, parent=0x0, extension=0x7f3198fface0) at ldap/servers/slapd/factory.c:348
#3  0x00007f319e54edf4 in operation_done (op=op@entry=0x7f3197f33b50, conn=0x0) at ldap/servers/slapd/operation.c:209
#4  0x00007f319e54ef16 in operation_free (op=0x7f3197f33b50, conn=<optimized out>) at ldap/servers/slapd/operation.c:241
#5  0x00007f319e551974 in pblock_done (pb=0x7f3197f33b40) at ldap/servers/slapd/pblock.c:210
#6  0x00007f319e551a5a in slapi_pblock_destroy (pb=<optimized out>) at ldap/servers/slapd/pblock.c:233
#7  0x00007f319e5649a1 in slapi_search_internal_get_entry
    (dn=0x7f3197f6d0f0, attrs=attrs@entry=0x0, ret_entry=ret_entry@entry=0x7f316c7fc7e0, component_identity=0x7f319c912fc0) at ldap/servers/slapd/plugin_internal_op.c:889
#8  0x00007f3199fd8a4b in ipauuid_pre_op (pb=0x7f3197f33ae0, modtype=4) at /usr/src/debug/freeipa-4.10.1-4.fc38.x86_64/daemons/ipa-slapi-plugins/ipa-uuid/ipa_uuid.c:827
#9  0x00007f319e55f3fc in plugin_call_func (list=0x7f319c90ad00, operation=operation@entry=421, pb=pb@entry=0x7f3197f33ae0, call_one=call_one@entry=0)
    at ldap/servers/slapd/plugin.c:2001
#10 0x00007f319e55f759 in plugin_call_list (pb=0x7f3197f33ae0, operation=421, list=<optimized out>) at ldap/servers/slapd/plugin.c:1944
#11 plugin_call_plugins (pb=0x7f3197f33ae0, whichfunction=421) at ldap/servers/slapd/plugin.c:414
#12 0x00007f319e54cb1d in op_shared_modify (pb=pb@entry=0x7f3197f33ae0, pw_change=pw_change@entry=0, old_pw=0x0) at ldap/servers/slapd/modify.c:1002
#13 0x00007f319e54e70e in modify_internal_pb (pb=0x7f3197f33ae0) at ldap/servers/slapd/modify.c:572
#14 0x00007f319e594c7c in modify_internal_entry (mods=0x7f316c7fce10, dn=0x7f317faecbb0 "cn=sidgen,cn=ipa-sidgen-task,cn=tasks,cn=config") at ldap/servers/slapd/task.c:772
#15 slapi_task_status_changed (task=0x7f317fa0f5e0) at ldap/servers/slapd/task.c:363
#16 0x00007f319e594f85 in slapi_task_status_changed (task=task@entry=0x7f317fa0f5e0) at ldap/servers/slapd/task.c:392
#17 0x00007f319e594fd2 in slapi_task_inc_progress (task=0x7f317fa0f5e0) at ldap/servers/slapd/task.c:148
#18 0x00007f3199fc20e1 in sidgen_task_thread (arg=0x7f317fa0f5e0) at /usr/src/debug/freeipa-4.10.1-4.fc38.x86_64/daemons/ipa-slapi-plugins/ipa-sidgen/ipa_sidgen_task.c:200
#19 0x00007f319e2ae907 in start_thread (arg=<optimized out>) at pthread_create.c:444
#20 0x00007f319e334870 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
(gdb) 
```

See the companion issue on IPA side: https://pagure.io/freeipa/issue/9406

**Comment by flo-renaud** [comment]
[slapd.coredump.gz](https://github.com/389ds/389-ds-base/files/11928958/slapd.coredump.gz)

**Comment by flo-renaud** [comment]
The tests were successful last week with 389-ds-base-2.4.1-202306260400git4b2f9ecd8.fc38.x86_64 and failed this week with 389-ds-base-2.4.1-202307020400gitf2c1e44b5.fc38.x86_64. The following commits happened in between:

* [Issue 5551 - Almost empty and not loaded ns-slapd high cpu load](https://github.com/389ds/389-ds-base/commit/f2c1e44b59c726091a7dcee6046daaeb490f0f77) - f2c1e44b59c726091a7dcee6046daaeb490f0f77
* [Issue 5755 - The Massive memory leaking on update operations (](https://github.com/389ds/389-ds-base/commit/b6cede516fe2620f577423a1beeee673a4e1444a)https://github.com/389ds/389-ds-base/pull/5803[)](https://github.com/389ds/389-ds-base/commit/b6cede516fe2620f577423a1beeee673a4e1444a) - b6cede516fe2620f577423a1beeee673a4e1444a

**Comment by mreynolds389** [comment]
Known issue.  It's a regression from a memory leak fix (#5755), we are working to fix it ASAP...

**Comment by flo-renaud** [comment]
Issue not reproduced any more with PR [2813](https://github.com/freeipa-pr-ci2/freeipa/pull/2813), fixed on 389ds side.
